hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RSProcedureDispatcher.java

```diff
@@ -306,6 +308,10 @@ private boolean scheduleForRetry(IOException e) {
           serverName, e.toString(), numberOfAttemptsSoFar);
         return false;
       }
+      if (isSaslError(e) && numberOfAttemptsSoFar == 0) {
```

Comment:
Does it matter how many attempts we have had so far if now we are getting a SASL error?

Reply:
Duo's point is that, if some attempts were already successful, it might have triggered region transition already and we might be in the middle of another sub-procedure when we encounter this.

I also think num of attempts should not matter as we won't be able to make any progress anyways, but then it takes our discussion back to the parent Jira HBASE-28048

Reply:
Yes, exactly as @virajjasani said, we need to make sure that the remote procedure has not been sent to the rs then we are safe to quit and choose another rs, otherwise the only safe way is to rely on SCP to tell us the rs is dead so we are safe to quit here.

So if we hit another error, like connection timed out the first time, then we are not sure whether we have already send the procedure to the rs, then no matter what the exceptions are in the following retries, we are not safe to quit.

In the real world, if authentication is not configured correctly, then it is likely we will get sasl error at the first try, so the code is enough to cover the problem here.

Reply:
@virajjasani Better to introduce a method may be called
```
boolean hasNotReachedRegionServerYet(IOException e)
```
In this method we could test for both sasl error and call queue too big, and also other exception types in the future. So we do not need to add extra condition every time when we want to add new exception type tests in scheduleForRetry method.

Reply:
done